s3transfer/utils.py

```diff
@@ -439,8 +439,8 @@ def enable_callback(self):
     def disable_callback(self):
         self._callbacks_enabled = False
 
-    def seek(self, where):
-        self._fileobj.seek(self._start_byte + where)
+    def seek(self, where, whence=0):
```

Comment:
So I think this implementation of seek will be a bit more tricky than the other ones. Mainly because the `ReadFileChunk` class has an additional abstraction that controls where you start and limits how far you can read into the file. Using the `self._start_byte` for the beginning of the file and `self._size` for the end of the file, this class ensures you cannot read before the beginning or after the end. So I think that passing in `whence` of non-zero breaks this abstraction. To fix it, I think it would be reasonable to:

1) Calculate the `where` value that is passed into `self._fileobj.seek()` based on the value of `whence` passed in along with the `self._start_byte` and `self._size`. So for `0`, where would be `self._start_byte + where`. For `1`, `self._amount_read + where`. For `2`, `self._size + where`.

2) Always pass `0` for `whence` into `self._fileobj.seek()`

3) Use the same calculated value in the progress callback and when setting `self._amount_read`

Let me know if that makes sense.

Reply:
Ping (kyleknap)!  This problem is still unresolved.  Is your proposed fix ready to merge (or post here as a patch for verification testing)?  Thanks!

Reply:
I took the liberty of handling the comments above. My PR is in #102.